src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= 50;
+    }
+
+    if (herbivore.health <= 0) {
+      const deadAnimalIndex = Animal.alive
+        .findIndex(({ name }) => name === herbivore.name);
+
+      Animal.alive.splice(deadAnimalIndex, 1);
+    }
```

Comment:
the problem is here. I have no idea why findIndex is not working but indexOf is working, but if you push animals not is Animal class< but instead in Carnivore and Herbivore classes,